### PR TITLE
Silence most warnings in test run

### PR DESCRIPTION
**Pull request by Ian-Foote**
For #91.

### CODE REVIEW

File: conman/redirects/migrations/0001_initial.py [hunk 1]
```diff
@@ -14,9 +14,9 @@ class Migration(migrations.Migration):
         migrations.CreateModel(
             name='RouteRedirect',
             fields=[
-                ('route_ptr', models.OneToOneField(serialize=False, auto_created=True, to='routes.Route', parent_link=True, primary_key=True)),
+                ('route_ptr', models.OneToOneField(serialize=False, auto_created=True, to='routes.Route', parent_link=True, primary_key=True, on_delete=models.CASCADE)),
                 ('permanent', models.BooleanField(default=False)),
-                ('target', models.ForeignKey(to='routes.Route', related_name='+')),
+                ('target', models.ForeignKey(to='routes.Route', related_name='+', on_delete=models.CASCADE)),
```
Comment (Ian-Foote) [inline conman/redirects/migrations/0001_initial.py]: This is the old default, so making it explicit by editing the migration is safe.

File: conman/redirects/models.py [hunk 2]
```diff
@@ -13,9 +14,11 @@ class RouteRedirect(Route):
     This model holds the data required to make that connection.
     """
     handler = handlers.RouteRedirectHandler.path()
-    target = models.ForeignKey('routes.Route', related_name='+')
+    target = models.ForeignKey('routes.Route', related_name='+', on_delete=models.CASCADE)
```
Comment (Ian-Foote) [inline conman/redirects/models.py]: This will be [required in Django 2.0](https://docs.djangoproject.com/en/1.10/releases/1.9/#foreignkey-and-onetoonefield-on-delete-argument).

File: conman/redirects/models.py [hunk 3]
```diff
@@ -13,9 +14,11 @@ class RouteRedirect(Route):
...
     permanent = models.BooleanField(default=False, blank=True)
 
+    objects = RouteManager()
```
Comment (Ian-Foote) [inline conman/redirects/models.py]: This seemed to me the cleanest option to handle [the inheritance changes planned for Django 2.0](https://docs.djangoproject.com/en/1.10/releases/1.10/#manager-use-for-related-fields-and-inheritance-changes).

I tried the `Meta.manager_inheritance_from_future=True` option, but that didn't play well with `makemigrations` on earlier Django versions.

We may want to explicitly document something around this.

File: conman/redirects/tests/test_models.py [hunk 4]
```diff
@@ -23,7 +24,8 @@ def test_fields(self):
 
     def test_target_self(self):
         """A RouteRedirect's target cannot be itself."""
-        redirect = ChildRouteRedirectFactory.create()
+        with ignore_django20_deprecation():
+            redirect = ChildRouteRedirectFactory.create()
```
Comment (Ian-Foote) [inline conman/redirects/tests/test_models.py]: This ignores two warnings related to `django-polymorphic`:

```
polymorphic/models.py:219: RemovedInDjango20Warning: Usage of field.rel has been deprecated. Use field.remote_field instead.
  super_to_sub_related_field = field_to_super.rel
django/utils/functional.py:35: RemovedInDjango20Warning: use_for_related_fields is deprecated, instead set Meta.base_manager_name on 'polymorphic.PolymorphicModel'.
  res = instance.__dict__[self.name] = self.func(instance)
```

File: conman/redirects/tests/test_views.py [hunk 5]
```diff
@@ -43,7 +43,8 @@ def test_access_redirect(self):
         """Accessing a RouteRedirect's url redirects to its target's url."""
         target = ChildRouteFactory.create()
         route = ChildRouteRedirectFactory.create(target=target)
-        response = self.client.get(route.url)
+        with ignore_django20_deprecation():
+            response = self.client.get(route.url)
```
Comment (Ian-Foote) [inline conman/redirects/tests/test_views.py]: This complains about the [`MIDDLEWARE` setting](https://docs.djangoproject.com/en/1.10/topics/http/middleware/#upgrading-middleware) that replaces `MIDDLEWARE_CLASSES`.

As far as I can tell, we can't switch until we drop support for 1.8 and 1.9.

File: conman/routes/managers.py [hunk 6]
```diff
@@ -6,6 +6,8 @@
 
 class RouteManager(PolymorphicManager):
     """Helpful methods for working with Routes."""
+    silence_use_for_related_fields_deprecation = True
```
Comment (Ian-Foote) [inline conman/routes/managers.py]: Suggested in: https://docs.djangoproject.com/en/1.10/releases/1.10/#manager-use-for-related-fields-and-inheritance-changes

File: conman/redirects/tests/test_models.py (hunk 4 above)
Comment (meshy) [inline conman/redirects/tests/test_models.py]: I'm very concerned about silencing warnings that we're not truly addressing, as it may hide issues.

If these warnings are raised from django-polymorphic, I'd rather see the warning that simply silence it.

File: conman/redirects/tests/test_views.py (hunk 5 above)
Comment (meshy) [inline conman/redirects/tests/test_views.py]: As above, if we're not actually fixing the issue, the warning is justified, and should remain.

File: conman/routes/handlers.py [hunk 7]
```diff
@@ -1,4 +1,7 @@
-from django.core.urlresolvers import resolve
+from conman.compat import ignore_django20_deprecation
+
+with ignore_django20_deprecation():
+    from django.core.urlresolvers import resolve
```
Comment (meshy) [inline conman/routes/handlers.py]: Rather than suppress the warning, could we instead do:

```python
try:
    from django.urls import resolve
except ImportError:
    from django.core.urlresolvers import resolve
```

...or alternatively add the import to the `compat` module, and import from there?

Comment (Ian-Foote) [inline conman/routes/handlers.py]: I considered that, but it messes with coverage.

Comment (meshy) [inline conman/routes/handlers.py]: Although, in particular because of the `#noqa`, I think I would be most comfortable with this in the `compat` module.

**Comment by meshy** [comment]
After discussion IRL: might be worth dropping django < 1.10 at this early stage to simplify this process.

File: conman/routes/managers.py (hunk 6 above)
Comment (meshy) [inline conman/routes/managers.py]: Now django < 1.10 is gone, is it worth considering `manager_inheritance_from_future=True`?

Comment (meshy) [inline conman/routes/managers.py]: After discussion, that doesn't appear to be related.

File: conman/routes/tests/test_handlers.py [hunk 8]
```diff
@@ -65,6 +66,7 @@ def setUp(self):
         class URLConfRoute(Route):
```
Comment (Ian-Foote) [inline conman/routes/tests/test_handlers.py]: Creating the same model repeatedly causes new warnings. I haven't tried to squelch them.

File: conman/redirects/models.py [hunk 9]
```diff
@@ -12,11 +12,13 @@ class RouteRedirect(Route):
 
     This model holds the data required to make that connection.
     """
-    target = models.ForeignKey('routes.Route', related_name='+')
+    target = models.ForeignKey('routes.Route', related_name='+', on_delete=models.CASCADE)
     permanent = models.BooleanField(default=False)
 
     view = views.RouteRedirectView.as_view()
 
+    base_objects = models.Manager()
```
Comment (meshy) [inline conman/redirects/models.py]: It's not so much that we don't *want* to inherit it -- more that I'm not sure if we *need* to. I'm very much looking forward to the behaviour coming in django 2.0.

File: conman/routes/tests/test_handlers.py (hunk 8 above)
Comment (meshy) [inline conman/routes/tests/test_handlers.py]: Do you think moving the definition into the `setupclass` (or even a `tests.models` module) would make sense?

Comment (Ian-Foote) [inline conman/routes/tests/test_handlers.py]: Makes sense to me.

**Comment by meshy** [comment]
Thanks!
